images/win/scripts/Installers/Install-JavaTools.ps1

```diff
@@ -3,29 +3,20 @@
 ##  Desc:  Install various JDKs and java tools
 ################################################################################
 
+Import-Module -Name ImageHelpers -Force
+
 # Download the Azul Systems Zulu JDKs
 # See https://www.azul.com/downloads/azure-only/zulu/
-$azulJDK7Uri = 'https://repos.azul.com/azure-only/zulu/packages/zulu-7/7u232/zulu-7-azure-jdk_7.31.0.5-7.0.232-win_x64.zip'
-$azulJDK8Uri = 'https://repos.azul.com/azure-only/zulu/packages/zulu-8/8u222/zulu-8-azure-jdk_8.40.0.25-8.0.222-win_x64.zip'
-$azulJDK11Uri = 'https://repos.azul.com/azure-only/zulu/packages/zulu-11/11.0.4/zulu-11-azure-jdk_11.33.15-11.0.4-win_x64.zip'
-
-cd $env:TEMP
-
-Invoke-WebRequest -UseBasicParsing -Uri $azulJDK7Uri -OutFile azulJDK7.zip
-Invoke-WebRequest -UseBasicParsing -Uri $azulJDK8Uri -OutFile azulJDK8.zip
-Invoke-WebRequest -UseBasicParsing -Uri $azulJDK11Uri -OutFile azulJDK11.zip
-
-# Expand the zips
-Expand-Archive -Path azulJDK7.zip -DestinationPath "C:\Program Files\Java\" -Force
-Expand-Archive -Path azulJDK8.zip -DestinationPath "C:\Program Files\Java\" -Force
-Expand-Archive -Path azulJDK11.zip -DestinationPath "C:\Program Files\Java\" -Force
-
-# Deleting zip folders
-Remove-Item -Recurse -Force azulJDK7.zip
-Remove-Item -Recurse -Force azulJDK8.zip
-Remove-Item -Recurse -Force azulJDK11.zip
-
-Import-Module -Name ImageHelpers -Force
+$azulJDKURLs = @(
+    'https://repos.azul.com/azure-only/zulu/packages/zulu-7/7u232/zulu-7-azure-jdk_7.31.0.5-7.0.232-win_x64.zip',
+    'https://repos.azul.com/azure-only/zulu/packages/zulu-8/8u222/zulu-8-azure-jdk_8.40.0.25-8.0.222-win_x64.zip',
+    'https://repos.azul.com/azure-only/zulu/packages/zulu-11/11.0.4/zulu-11-azure-jdk_11.33.15-11.0.4-win_x64.zip'
+)
+
+$azulJDKURLs | ForEach-Object {
+    $archivePath = Start-DownloadWithRetry -Url $_ -Name $_.Split("/")[-1]
```

Comment:
Use `[IO.Path]::GetFileName()` instead of `$_.Split("/")[-1]`
Also, we can get rid of `Name` parameter because it is optional, I think

Reply:
Done, thanks! But, `Name` is mandatory parameter.